app/models/submission.rb

```diff
@@ -0,0 +1,40 @@
+class Submission < ApplicationRecord
+  belongs_to :user
+
+  has_many :comments, class_name: "Submission", foreign_key: :post_id
+  belongs_to :post, class_name: "Submission", optional: true
+
+  validates :user, presence: true
+  validates :title, :url, presence: true, if: :url_submission?
+
+  validate :text_submission
+  validate :at_least_one_field_present
+
+  def limit_text
+    ary = self.text.chars.shift(80)
+    shift_pos = 0
+    ary.reverse.each_with_index do |val, index|
+      shift_pos = index + 1 if val.blank?
+      break if shift_pos > 0
+    end
+    "#{ary.shift(ary.count - shift_pos).join}..."
+  end
```

Comment:
Could https://apidock.com/rails/String/truncate work?

Reply:
Ah, of course a truncate method exists - I never looked! But, I like the way my method cuts the string on a blank space between words. I'll keep it but glad to know about truncate for future projects.

Reply:
Ah cool. I wonder if the `separate` option gets you something similar. It looks like you can do something like:

```ruby
"The quick brown fox jumps over the fence".truncate(25, separator: /\s/)
# "The quick brown fox..."
```

Reply:
Perfect! I'd glanced at that link too quickly. 

Man, I need to stop spending time writing methods that Ruby and Rails already provide! At least now I'm inspired to pick up more new methods from the source code.